src/anchorfm-pupeteer/index.js

```diff
@@ -9,19 +9,19 @@ function addUrlToDescription(youtubeVideoInfo) {
 
 async function setPublishDate(page, navigationPromise, date) {
   console.log('-- Setting publish date');
-  const publishDateButtonSelector = '//span[contains(text(),"Publish date:")]/following-sibling::button';
-  const [publishDateButton] = await page.$x(publishDateButtonSelector);
-  await publishDateButton.click();
+  await clickXpath(page,
+    '//span[contains(text(),"Publish date:")]/following-sibling::button',
+    env.UPLOAD_TIMEOUT)
   await navigationPromise;
 
   await resetDatePickerToSelectYears(page, navigationPromise);
   await selectYearInDatePicker(page, navigationPromise, date.year);
   await selectMonthInDatePicker(page, navigationPromise, date.month);
   await selectDayInDatePicker(page, navigationPromise, date.day);
 
-  const confirmButtonSelector = '//span[contains(text(),"Confirm")]/parent::button';
-  const [confirmButton] = await page.$x(confirmButtonSelector);
-  await confirmButton.click();
+  await clickXpath(page,
```

Comment:
I think you miss a couple of places where you can apply this refactor, unless I am missing something.

Here are the places:

`await datePickerSwitchButton.click();`

`await tdYear.click();`

`await tdMonth.click();`

`await tdDay.click();`

`await contentTypeLabel.click();`

`await saveEpisodeArtButton.click();`

`await saveDraftOrPublishOrScheduleButton.click();`